src/renderer/base/RenderSettings.cpp

```diff
@@ -112,6 +112,13 @@ COLORREF RenderSettings::GetColorTableEntry(const size_t tableIndex) const
     return _colorTable.at(tableIndex);
 }
 
+// Routine Description:
+// - Restores a range of color table entries to the values saved in SaveDefaultSettings.
+void RenderSettings::RestoreDefaultColorTableEntries(const size_t startIndex, const size_t count)
+{
+    std::copy_n(&_defaultColorTable.at(startIndex), count, &_colorTable.at(startIndex));

```

Comment:
This seems like it could use some bounds checks. Perhaps just a `std::min` together with a soft `assert` though? Not sure... Alternative we could make it two explicit reset-256-indexed and reset-specific-index functions.

Reply:
So, yeah. I feel weird about this function.

Everywhere else in this file we use .at() and allow the SafeExecute handler to catch the exception. No bounds checks, no worries.

I was so averse to adding members to ITermDispatch that I forgot I could add members to RenderSettings with no real cost.